gax-java/gax-httpjson/src/test/java/com/google/api/gax/httpjson/RetryingTest.java

```diff
@@ -263,7 +297,12 @@ void retryOnUnexpectedException() {
     UnaryCallable<Integer, Integer> callable =
         HttpJsonCallableFactory.createUnaryCallable(
             callInt, callSettings, httpJsonCallSettings, clientContext);
-    ApiException exception = assertThrows(ApiException.class, () -> callable.call(initialRequest));
+    // Should an unexpected RuntimeException be received as an UnknownException?
+    UnknownException exception =
+        assertThrows(UnknownException.class, () -> callable.call(initialRequest));
```

Comment:
Might be missing some context, but why does it need to be received as an UnknownException? Was there a runtime issue with receiving it as an ApiException?

Reply:
@diegomarquezp to share the full stack trace of both exceptions for further analysis (from discussion with @blakeli0 @lqiu96 and @zhumin8)

Reply:
Hmm, I tried assertThrows locally with ApiException and it passed: `ApiException exception = assertThrows(ApiException.class, () -> callable.call(initialRequest));`

Reply:
Hmm, yes that's true. I can't figure how that error came up then. I'll modify the description. Thank you for double checking this.